cmd/migration-managerd/internal/api/api_source.go

```diff
@@ -357,11 +369,21 @@ func sourcePut(d *Daemon, r *http.Request) response.Response {
 		return response.SmartError(fmt.Errorf("Failed commit transaction: %w", err))
 	}
 
-	// Trigger a scan of this new source for instances.
-	err = d.syncOneSource(d.ShutdownCtx, src)
+	d.checkSourceConnectivity()
```

Comment:
In this case, since you're ignoring the connectivity error (and therefore the sync error) if you update source A to point to some source B, but then the connectivity/sync silently fails, when you go to create a batch, it will register the old source's instances.